src/VersionTool/Program.cs

```diff
@@ -225,13 +253,324 @@ private static int OnList(CommandOption pathOption)
+            SaveJObject(patchConfigPath, patchConfig);
+
+            return 0;
+        }
+
+        private static void UpdatePackagesInRepo(Repository repo, string directory, IEnumerable<Package> packages, IEnumerable<Rule> rules)
+        {
+            foreach (var package in EnumerateProjects(new[] { Path.Combine(repo.Path, "src") }))
+            {
+                var unUpdatePackage = packages.Where(p => p.Name == package.Name && p.CurrentVersion == p.NewVersion).SingleOrDefault();
+                if (unUpdatePackage != null)
+                {
+                    Console.WriteLine($"{package.Name} is a src project and has not been updated in this patch, updating version.");
+
+                    var rule = GetRuleForPackage(package, rules);
+
+                    if (rule == null)
+                    {
+                        throw new Exception($"No update rule found for package {package.Name} {package.Version}.");
+                    }
+
+                    // update version
+                    OnUpdateVersion(
+                        new List<string>(new[] { package.ProjectDirectory }),
+                        new List<string>(new[] { rule.CurrentVersion }),
+                        rule.NewVersion);
+
+                    // update dependency
+                    OnUpdateDependency(
+                        new List<string>(new[] { directory }),
+                        new List<string>(new[] { rule.CurrentVersion }),
+                        package.Name,
+                        rule.NewVersion);
+
+                    // update entry in packages
+                    unUpdatePackage.NewVersion = rule.NewVersion;
+                }
+                else
+                {
+                    Console.WriteLine($"{package.Name} is a src project but has been updated, skipping.");
+                }
+            }
+        }
+
+        private static void BuildRepoGraph(List<Repository> repos)
+        {
+            var includedSubDirs = new[] { "src", "test", "samples", "tools" };
+
+            foreach (var repo in repos)
+            {
+                foreach (var subdir in includedSubDirs)
+                {
+                    var repoSubDir = Path.Combine(repo.Path, subdir);
+                    if (Directory.Exists(repoSubDir))
+                    {
+                        if (subdir == "src")
+                        {
+                            repo.Packages.UnionWith(Directory.EnumerateDirectories(repoSubDir).Select(path => new DirectoryInfo(path).Name));
+                        }
+
+                        foreach (var projectJsonPath in Directory.GetFiles(repoSubDir, "project.json", SearchOption.AllDirectories))
+                        {
+                            AddDependencies(projectJsonPath, repo.PackageDependencies);
+                        }
+                    }
+                }
+
+                repo.PackageDependencies.ExceptWith(repo.Packages);
```

Comment:
What do you mean? you need to add the dependencies that are contained in all the `src` dependencies but you don't want project dependencies in the same repo. Those are not useful.